genie/process_functions.py

```diff
@@ -948,3 +948,35 @@ def create_new_fileformat_table(
         "newdb_mappingdf": newdb_mappingdf,
         "moved_ent": moved_ent,
     }
+
+
+def create_missing_columns(dataset: pd.DataFrame, schema: dict) -> pd.Series:
+    """Creates and fills missing columns with the relevant NA value for the
+        given data type. Note that special handling had to occur for
+        allowing NAs in integer based columns in pandas by converting
+        the integer column into the Int64 (pandas nullable integer data type)
+
+    Args:
+        dataset (pd.DataFrame): input dataset to fill missing columns for
+        schema (dict): the expected schema {column_name(str): data_type(str)}
+            for the input dataset
+
+    Returns:
+        pd.Series: updated dataset
+    """
+    missing_values = {
+        "string": "",
+        "integer": None,
+        "float": float("nan"),
+        "boolean": None,
+    }
+    for column, data_type in schema.items():
+        if column not in dataset.columns:
+            dataset = dataset.assign(**{column: missing_values[data_type]})
+
+        # only way to preserve NAs for these specific dtype columns
```

Comment:
I agree, I'm happy for you to comment out this function since you wrote it and revisit, but I think we should have a set schema for the MAF.

Definitely look at the simplest solution with least amount of tech debt.